test/e2e/kcp_upgrade.go

```diff
@@ -50,8 +49,7 @@ func KCPUpgradeSpec(ctx context.Context, inputGetter func() KCPUpgradeSpecInput)
 		input         KCPUpgradeSpecInput
 		namespace     *corev1.Namespace
 		cancelWatches context.CancelFunc
-		cluster       *clusterv1.Cluster
-		controlPlane  *controlplanev1.KubeadmControlPlane
+		result        *clusterctl.ApplyClusterTemplateAndWaitResult
```

Comment:
What about?
```suggestion
		clusterReosurces        *clusterctl.ApplyClusterTemplateAndWaitResult
```
(result does not express what the variables really is)